IdentityCore/src/requests/sdk/msal/MSIDDefaultTokenResponseValidator.m

```diff
@@ -91,6 +91,15 @@ - (BOOL)validateAccount:(MSIDAccountIdentifier *)accountIdentifier
     {
         MSID_LOG_WITH_CORR_PII(MSIDLogLevelError, correlationID, @"Different account was returned from the server. Original account %@, returned account %@", MSID_PII_LOG_TRACKABLE(accountIdentifier.uid), MSID_PII_LOG_TRACKABLE(tokenResult.account.accountIdentifier.uid));
         
+        // Probably the account was deleted and re-created with the same upn and request contained old cached user's object id
+        if (tokenResult.account.accountIdentifier.utid != nil &&
```

Comment:
Add a unit test?

Reply:
Added. Will wait for Jason's input before merging as he had opened a fix earlier but abandoned it.